File: python/AllocatedPiece1D.py
import numpy as np

class AllocatedPiece1D:
    """ /**
     * A class representing a piece allocated on a 1-dimensional cake to an agent with a given value-function.
     *
     * @author Erel Segal-Halevi
     * @since 2016-11
     */"""

     def __init__(self, valueFunction, iFrom, iTo):
         """ /**
         *  Initialize a 1-dimensional allocated piece based on a value function.
         *  @param valueFunction a ValueFunction1D.
         *  @param iFrom (float) start of allocation.
         *  @param iTo (float) end of allocation.
         */ """
    	if iFrom is None: iFrom = 0
    	if iTo is None:   iTo = valueFunction.values.length

    	self.valueFunction = valueFunction
    	self.iFrom = iFrom
    	self.iTo = iTo

    def getCut(self, value):
		"""
		The current agent decides where to cut his current allocation
		such that the cut piece will have the specified value.
		"""
		return self.valueFunction.getCut(self.iFrom, value)

    def getValue(self):
		"""
		The current agent evaluates his own piece.
		"""
		return self.valueFunction.getValue(self.iFrom, self.iTo)

    def getValueOfOther(self, other):
        """
        The current agent evaluates a piece allocated to another agent.
        """
        return self.valueFunction.getValue(other.iFrom, other.iTo);

    def getRelativeValue(self):
        """
        The current agent evaluates his own piece relative to the entire cake.
        """
        return self.valueFunction.getRelativeValue(self.iFrom, self.iTo)

File: lib/AllocatedPiece1D.js
/**
 * A class representing a piece allocated on a 1-dimensional cake.
 *
 * @author Erel Segal-Halevi
 * @since 2014-08
 * @deprecated by AllocatedPiece1D.py
 */

/**
 *  Initialize a 1-dimensional allocated piece based on a value function.
 *  @param valueFunction a ValueFunction1D.
 *  @param iFrom (float) start of allocation.
 *  @param iTo (float) end of allocation.
 */
var AllocatedPiece1D = function(valueFunction,iFrom,iTo) {
	if (isNaN(iFrom)) throw new Error("iFrom is NaN")
	if (isNaN(iTo)) throw new Error("iTo is NaN")
	if (!iFrom) iFrom = 0;
	if (!iTo) iTo = valueFunction.values.length;

	this.valueFunction = valueFunction;
	this.iFrom = iFrom;
	this.iTo = iTo;
}


/*** factory methods ****/

AllocatedPiece1D.fromValueFunction = function(valueFunction) {
	return new AllocatedPiece1D(valueFunction, 0, valueFunction.values.length);
}

AllocatedPiece1D.fromValueFunctionAndBounds = function(valueFunction,iFrom,iTo) {
	return new AllocatedPiece1D(valueFunction, iFrom, iTo);
}



/*** query methods ****/

/**
 * Cut query.
 * @param value what the piece value should be.
 * @return where the piece should end.
 */
AllocatedPiece1D.prototype.getCut = function(value) {
	return this.valueFunction.getCut(this.iFrom, value);
}

AllocatedPiece1D.prototype.getValue = function() {
	return this.valueFunction.getValue(this.iFrom, this.iTo);
}

AllocatedPiece1D.prototype.getValueOfOther = function(other) {
	return this.valueFunction.getValue(other.iFrom, other.iTo);
}

AllocatedPiece1D.prototype.getRelativeValue = function() {
	return this.valueFunction.getRelativeValue(this.iFrom, this.iTo);
}

module.exports = AllocatedPiece1D;
